onnx/optimizer/passes/fuse_consecutive_concats.h

```diff
@@ -39,8 +56,9 @@ struct FuseConsecutiveConcats final : public PredicateBasedPass {
           cur_input_node->hasAttribute(kaxis) &&
           cur_input_node->i(kaxis) == concat_node->i(kaxis)) {
         transform_ran = true;
-        for (auto value : cur_input_node->inputs()) {
-          concat_node->addInput(value);
+        for (size_t j = 0; j < cur_input_node->inputs().size(); j++) {
+          Value* value = cur_input_node->input(j);
+          insertInput(concat_node, i + 1 + j, value);
```

Comment:
Could you further explain why using `i+1+j` as the index here?

Reply:
> Could you further explain why using `i+1+j` as the index here?

It inserts n inputs of `cur_input_node` at index i+1~i+1+(n-1), and remove `cur_input_node` at index i. As a result, `cur_input_node` is replaced by its inputs **inplace**, instead of the wrong way: always append its inputs at the end.

Reply:
Can you please put this clarification as comments in the source code?

btw, do you mind adding some test cases to avoid regression?

Reply:
@linkerzhang Thanks! I have added the comments and updated the test